app/views/helpers/TimelineEvents.scala

```diff
@@ -49,18 +49,30 @@ class TimelineEvents @Inject() (
   link: link,
   linkButton: linkButton,
   paragraphBody: paragraphBody,
-  secureMessagingInboxConfig: SecureMessagingInboxConfig,
   sfusConfig: SfusConfig,
+  secureMessagingInboxConfig: SecureMessagingInboxConfig,
+  declarationAmendmentsConfig: DeclarationAmendmentsConfig,
   uploadFilesPartialForTimeline: upload_files_partial_for_timeline
 ) {
   def apply(submission: Submission)(implicit messages: Messages): Seq[TimelineEvent] = {
 
-    val notificationEvents = createNotificationEvents(submission)
+    val allNotificationEvents = createNotificationEvents(submission)
+    val hasExternalAmendments = allNotificationEvents.exists(_.requestType == ExternalAmendmentRequest)
```

Comment:
This is a little confusing. It looks like it is saying "If we have not received any ExternalAmendmentRequests" then display all notifications. I can see digging into the code that no ExternalAmendmentRequests events will be generated if the feature flag is disabled, but its not obvious from this code block.

Could you instead maybe always add both the ExternalAmendmentRequests and the SubmissionRequest notification of type AMEND and then filter them here based on the feature flag? So, if disabled filter out the ExternalAmendmentRequests events, if enabled filter out the SubmissionRequest notification? That would be more readable IMO :thinking: